File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/cloudtrail/src/main/java/com/example/cloudtrail/PutEventSelectors.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/
package com.example.cloudtrail;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.cloudtrail.CloudTrailClient;
import software.amazon.awssdk.services.cloudtrail.model.CloudTrailException;
import software.amazon.awssdk.services.cloudtrail.model.PutEventSelectorsRequest;
import software.amazon.awssdk.services.cloudtrail.model.EventSelector;
import java.util.ArrayList;
import java.util.List;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */

public class PutEventSelectors {

    public static void main(String[] args) {

       final String USAGE = "\n" +
                "Usage:\n" +
                "    <trailName> \n\n" +
                "Where:\n" +
                "    trailName - the name of the trail. \n" ;

        if (args.length != 1) {
            System.out.println(USAGE);
            System.exit(1);
        }

        String trailName = args[0];
        Region region = Region.US_EAST_1;
        CloudTrailClient cloudTrailClient = CloudTrailClient.builder()
                .region(region)
                .build();

        setSelector(cloudTrailClient, trailName);
        cloudTrailClient.close();
    }

    public static void setSelector(CloudTrailClient cloudTrailClientClient, String trailName) {

        try {
            EventSelector selector = EventSelector.builder()
                .readWriteType("All")
                .build();

            List<EventSelector> selList = new ArrayList<>();
            selList.add(selector);

            PutEventSelectorsRequest selectorsRequest = PutEventSelectorsRequest.builder()
                .trailName(trailName)
                .eventSelectors(selList)
                .build();

            cloudTrailClientClient.putEventSelectors(selectorsRequest);

        } catch (CloudTrailException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
   }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/cloudtrail/src/main/kotlin/com/kotlin/cloudtrail/PutEventSelectors.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.cloudtrail

import aws.sdk.kotlin.services.cloudtrail.CloudTrailClient
import aws.sdk.kotlin.services.cloudtrail.model.EventSelector
import aws.sdk.kotlin.services.cloudtrail.model.PutEventSelectorsRequest
import aws.sdk.kotlin.services.cloudtrail.model.ReadWriteType
import kotlin.system.exitProcess

suspend fun main(args: Array<String>) {

    val usage = """

    Usage:
        <trailName>  

    Where:
        trailName - the name of the trail. 
    """

     if (args.size != 1) {
         println(usage)
         exitProcess(0)
      }

    val trailName = args[0]
    setSelector(trailName)
    }

suspend fun setSelector(trailNameVal: String?) {

            val selector = EventSelector {
                readWriteType = ReadWriteType.fromValue("All")
            }

            CloudTrailClient { region = "us-east-1" }.use { cloudTrail ->
                cloudTrail.putEventSelectors(PutEventSelectorsRequest {
                    trailName = trailNameVal
                    eventSelectors = listOf(selector)
                } )
            }
    }
